noxfile.py

```diff
@@ -33,7 +33,7 @@ def pytest_coverage(session: Session) -> None:
 @nox.session(python=PYTHON_VERSIONS[0])  # type: ignore[misc]
 def minimum_versions(session: Session) -> None:
     session.install(
-        "pandas==0.25.3",
+        "pandas==1.1.5",
```

Comment:
@MarcoGorelli I made this change because tests are not passing. Pyarrow does not support `pandas< 1.0.0`.

Tests were broken by merging #404

Reply:
could you clarify please? looks like that PR (and the associated merge to main) pass CI https://github.com/narwhals-dev/narwhals/commits/main/ ?

Reply:
These are some of the errors I see when running test locally on latest version of main:

```text
UNEXPECTED EXCEPTION: AttributeError("'DataFrame' object has no attribute 'convert_dtypes'")
Traceback (most recent call last):
  File "/usr/lib/python3.8/doctest.py", line 1336, in __run
    exec(compile(example.source, filename, "single",
  File "<doctest narwhals.utils.maybe_convert_dtypes[6]>", line 1, in <module>
  File "/workspaces/narwhals/narwhals/utils.py", line 261, in maybe_convert_dtypes
    df_any._dataframe._dataframe.convert_dtypes(*args, **kwargs)
  File "/workspaces/narwhals/.nox/minimum_versions/lib/python3.8/site-packages/pandas/core/generic.py", line 5179, in __getattr__
    return object.__getattribute__(self, name)
AttributeError: 'DataFrame' object has no attribute 'convert_dtypes'
/workspaces/narwhals/narwhals/utils.py:249: UnexpectedException
```


```text
UNEXPECTED EXCEPTION: ImportError('pyarrow requires pandas 1.0.0 or above, pandas 0.25.3 is installed')
Traceback (most recent call last):
  File "/usr/lib/python3.8/doctest.py", line 1336, in __run
    exec(compile(example.source, filename, "single",
  File "<doctest narwhals.series.Series.to_pandas[8]>", line 1, in <module>
  File "/workspaces/narwhals/narwhals/translate.py", line 400, in wrapper
    result = func(*args, **kwargs)
  File "<doctest narwhals.series.Series.to_pandas[6]>", line 3, in func
  File "/workspaces/narwhals/narwhals/series.py", line 1209, in to_pandas
    return self._series.to_pandas()
  File "/workspaces/narwhals/.nox/minimum_versions/lib/python3.8/site-packages/polars/series/series.py", line 4434, in to_pandas
    else self.to_arrow().to_pandas(**kwargs)
  File "pyarrow/array.pxi", line 830, in pyarrow.lib._PandasConvertible.to_pandas
  File "pyarrow/array.pxi", line 1426, in pyarrow.lib.Array._to_pandas
  File "pyarrow/array.pxi", line 1649, in pyarrow.lib._array_like_to_pandas
  File "pyarrow/pandas-shim.pxi", line 103, in pyarrow.lib._PandasAPIShim.series
  File "pyarrow/pandas-shim.pxi", line 96, in pyarrow.lib._PandasAPIShim._check_import
  File "pyarrow/pandas-shim.pxi", line 65, in pyarrow.lib._PandasAPIShim._import_pandas
ImportError: pyarrow requires pandas 1.0.0 or above, pandas 0.25.3 is installed
/workspaces/narwhals/narwhals/series.py:1203: UnexpectedException
```

```text
______________________________________________ [doctest] narwhals.expression.ExprDateTimeNamespace.total_seconds ______________________________________________
2635 
2636             We define a dataframe-agnostic function:
2637 
2638             >>> @nw.narwhalify
2639             ... def func(df):
2640             ...     return df.with_columns(a_total_seconds=nw.col("a").dt.total_seconds())
2641 
2642             We can then pass either pandas or Polars to `func`:
2643 
2644             >>> func(df_pd)
Differences (unified diff with -expected +actual):
    @@ -1,3 +1,3 @@
    -                       a  a_total_seconds
    -0        0 days 00:00:10               10
    -1 0 days 00:00:20.040000               20
    +                a  a_total_seconds
    +0        00:00:10               10
    +1 00:00:20.040000               20

/workspaces/narwhals/narwhals/expression.py:2644: DocTestFailure
```

They all appear when running doctest with minimum versions. However, CICD passes just fine, so I am concerned there is a misconfiguration somewhere, probably missing a couple of `skipif`s. The error clearly states that `Pyarrow` requires `pandas >=1.0.0`

Reply:
@MarcoGorelli I believe I found the causes:

1. The `minimum_versions` is running `pandas==0.25.3` but wihtout the `--doctest-modules` - [Source](https://github.com/narwhals-dev/narwhals/blob/cc2c1f11983f48122d102d22382468192c13737e/.github/workflows/extremes.yml#L9-L34)
2. The `pretty_old_versions` is running `--doctest-modules` but without `pandas==0.25.3`, it is using `pandas==1.1.5` - [Source](https://github.com/narwhals-dev/narwhals/blob/cc2c1f11983f48122d102d22382468192c13737e/.github/workflows/extremes.yml#L36-L63)

Was this made purposely? If so, the noxfile and the tests do not accurately reflect what the CICD does

Reply:
ah thanks - I don't use nox myself, another dev added it, it should probably be updated